test/jdk/javax/swing/JRootPane/bug4403624.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2001, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4403624
+ * @summary Tests JRootPane layout with invisible menubar
+ * @key headful
+ * @run main bug4403624
+ */
+
+import java.awt.Color;
+import java.awt.Container;
+import java.awt.FlowLayout;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.SwingUtilities;
+
+public class bug4403624 {
+    private static JFrame f;
+    private static Container c;
+    private static JButton b;
+    private static volatile Point p;
+    private static final int OFFSET = 2;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                f = new JFrame("bug4403624 Test");
+                JMenuBar mbar;
+                mbar = new JMenuBar();
+                mbar.add(new JMenu("Menu"));
+                f.setJMenuBar(mbar);
+                b = new JButton("Hide Menu");
+                b.addActionListener(e -> mbar.setVisible(false));
+                c = f.getContentPane();
+                c.setLayout(new FlowLayout());
+                c.setBackground(Color.GREEN);
+                c.add(b);
+                f.pack();
+                f.setLocationRelativeTo(null);
+                f.setAlwaysOnTop(true);
+                f.setVisible(true);
+            });
+
+            Robot r = new Robot();
+            r.setAutoDelay(200);
+            r.waitForIdle();
+            r.delay(1000);
+
+            SwingUtilities.invokeAndWait(() -> p = b.getLocationOnScreen());
+
+            r.mouseMove(p.x + (b.getWidth() / 2), p.y + (b.getHeight() / 2));
```

Comment:
you called the b.getLocationOnScreen() on EDt but what about b.getWidth() and b.getHeight()

Reply:
Good point. I was previously advised to remove a `Point.move` call in favor of this, and I didn't put it in the EDT when doing so. Thanks, updated!